pkgs/by-name/vl/vlc/package.nix

```diff
@@ -1,106 +1,110 @@
-{ lib
-, SDL
-, SDL_image
-, a52dec
-, alsa-lib
-, autoreconfHook
-, avahi
-, curl
-, dbus
-, faad2
-, fetchpatch
-, fetchurl
-, ffmpeg
-, flac
-, fluidsynth
-, freefont_ttf
-, freetype
-, fribidi
-, genericUpdater
-, gnutls
-, libSM
-, libXext
-, libXinerama
-, libXpm
-, libXv
-, libXvMC
-, libarchive
-, libass
-, libbluray
-, libcaca
-, libcddb
-, libdc1394
-, libdvbpsi
-, libdvdnav
-, libebml
-, libgcrypt
-, libgpg-error
-, libjack2
-, libkate
-, libmad
-, libmatroska
-, libmicrodns
-, libmodplug
-, libmtp
-, liboggz
-, libopus
-, libplacebo_5
-, libpulseaudio
-, libraw1394
-, librsvg
-, libsForQt5
-, libsamplerate
-, libspatialaudio
-, libssh2
-, libtheora
-, libtiger
-, libupnp
-, libv4l
-, libva
-, libvdpau
-, libvorbis
-, libxml2
-, live555
-, lua5
-, mpeg2dec
-, ncurses
-, perl
-, pkg-config
-, pkgsBuildBuild
-, protobuf
-, removeReferencesTo
-, samba
-, schroedinger
-, speex
-, srt
-, stdenv
-, systemd
-, taglib
-, unzip
-, wayland
-, wayland-protocols
-, wrapGAppsHook3
-, writeShellScript
-, xcbutilkeysyms
-, zlib
-
-, chromecastSupport ? true
-, jackSupport ? false
-, onlyLibVLC ? false
-, skins2Support ? !onlyLibVLC
-, waylandSupport ? true
-, withQt5 ? true
+{
+  lib,
+  SDL,
+  SDL_image,
+  a52dec,
+  alsa-lib,
+  autoreconfHook,
+  avahi,
+  curl,
+  dbus,
+  faad2,
+  fetchpatch,
+  fetchurl,
+  ffmpeg,
+  flac,
+  fluidsynth,
+  freefont_ttf,
+  freetype,
+  fribidi,
+  genericUpdater,
+  gnutls,
+  libSM,
+  libXext,
+  libXinerama,
+  libXpm,
+  libXv,
+  libXvMC,
+  libarchive,
+  libass,
+  libbluray,
+  libcaca,
+  libcddb,
+  libdc1394,
+  libdvbpsi,
+  libdvdnav,
+  libebml,
+  libgcrypt,
+  libgpg-error,
+  libjack2,
+  libkate,
+  libmad,
+  libmatroska,
+  libmicrodns,
+  libmodplug,
+  libmtp,
+  liboggz,
+  libopus,
+  libplacebo_5,
+  libpulseaudio,
+  libraw1394,
+  librsvg,
+  libsForQt5,
+  libsamplerate,
+  libspatialaudio,
+  libssh2,
+  libtheora,
+  libtiger,
+  libupnp,
+  libv4l,
+  libva,
+  libvdpau,
+  libvorbis,
+  libxml2,
+  live555,
+  lua5,
+  mpeg2dec,
+  ncurses,
+  perl,
+  pkg-config,
+  pkgsBuildBuild,
+  protobuf,
+  removeReferencesTo,
+  samba,
+  schroedinger,
+  speex,
+  srt,
+  stdenv,
+  systemd,
+  taglib,
+  testers,
+  unzip,
+  wayland,
+  wayland-protocols,
+  wrapGAppsHook3,
+  writeShellScript,
+  xcbutilkeysyms,
+  zlib,
+  # Boolean flags
+  enableLive555 ? (!stdenv.hostPlatform.isAarch && !onlyLibVLC),
+  chromecastSupport ? true,
+  enableLibmicrodns ? chromecastSupport,
+  enableProtobuf ? chromecastSupport,
+  enableSystemd ? true,
```

Comment:
The code this enables is tiny, and disabling it does not remove systemd from the closure because (depending on how you count) there are at least 5 other paths.

Reply:
What are those five paths?

Reply:
For some reason I don't understand vlc ends up depending on the one with pname systemd-minimal-libs, and the other paths to that are vlc->qtbase->systemd, vlc->ffmpeg->SDL2->systemd, vlc->libmtp->libusb->systemd, vlc->samba->systemd (the regular variant this time), vlc->gtk+3->at-spi2-core->systemd (the regular one too). And that's with *really* conservative counting (only looking at the shortest paths and counting Qt as one).